e2e/cypress/integration/ui-smoke/ui-add-sample.spec.js

```diff
@@ -0,0 +1,57 @@
+/// <reference types="cypress" />
+import '../../support/commands';
+// import successResponse from '../../fixtures/successResponse.json';
+
+import { addFileActions } from '../../constants';
+
+const resizeObserverLoopErrRe = /ResizeObserver loop limit exceeded/;
+const projectName = 'IntTest - Add sample Project';
+
+describe('Sample addition/removal', () => {
+  // before each test:
+  //   1. set up the network intercepts
+  //   2. Log in into biomage
+  //   3. Visit data-management
+  beforeEach(() => {
+    cy.login();
+    cy.visit('/data-management');
+  });
+
+  afterEach(() => {
+    cy.removeSample();
+  });
+
+  // we have some kind of resize observer loop error that needs looking into
+  Cypress.on('uncaught:exception', (err) => {
+    if (resizeObserverLoopErrRe.test(err.message)) {
+      return false;
+    }
+    return true;
+  });
+
+  it('Adds a new sample correctly by drag and drop', () => {
+    cy.selectProject(projectName);
+
+    cy.addSample(addFileActions.DRAG_AND_DROP);
+
+    // Sample cell shows up
+    cy.get('.data-test-sample-name-cell').should('be.visible');
```

Comment:
I you look at `SamplesTable.jsx`, each of the rendered components inside the table is called `cell`, do you think I should call it `data-test-sample-name-table-cell` instead or something like that to avoid confusion?

Reply:
Maybe change to `data-test-samples-table-name-cell`?

Reply:
I think the cell part is just a particular implementation name for the table but it doesn't give any extra info to the outside. I would just call it '.data-test-sample-name'. Otherwise, it is too long and verbose. If there are other places where this name can be used (thus having potential conflicts) then I would resort to more complex selectors instead of longer names e.g. `cy.get('.data-test-samples-table > .data-test-sample-name')`. imho it's easier to see the structure and it becomes more readable instead of trying to put all the hierarchy structure in the name as agi suggested.

Regarding the actual selection, shouldn't it select the elements matching the given sample name instead of just all ay of them?  Something like (syntax might be incorrect):

`cy.get('.data-test-sample-name-cell').contains(sampleName).should('be.visible');`

Reply:
Im going with `data-test-samples-in-table` which seems descriptive and I think takes into account the suggestions in thecomments